app/Console/Commands/EsIndexScoresQueue.php

```diff
@@ -104,11 +90,66 @@ public function handle()
         $this->info("Queued {$this->total} scores in {$totalTime}s");
     }
 
+    private function parseOptions(): void
+    {
+        $query = Score::select('id');
+        $userId = get_int($this->option('user'));
+        if ($userId !== null) {
+            $query->where('user_id', $userId);
+        }
+
+        $doneParsingId = false;
+
+        $ids = $this->parseOptionIds();
+        if ($ids->count() > 0) {
+            $doneParsingId = true;
+            if ($userId === null) {
+                $this->ids = $ids->toArray();
+            } else {
+                $this->query = $query->whereKey($ids->toArray());
+            }
+        }
+
+        $from = get_int($this->option('from'));
+        if ($from !== null) {
+            if ($doneParsingId) {
+                throw new InvariantException('only one of the id parameters may be specified');
+            }
+            $doneParsingId = true;
+            $this->query = $query->where('id', '>', $from);
+        }
+
+        if ($this->option('all')) {
+            if ($doneParsingId) {
+                throw new InvariantException('only one of the id parameters may be specified');
+            }
+            $doneParsingId = true;
+            $this->query = $query;
+        }
+
+        if (!$doneParsingId) {
+            throw new InvariantException('id parameter must be specified');
+        }
+
+        $schema = presence($this->option('schema'));
+        if ($schema === null) {
+            $this->schemas = $this->search->getActiveSchemas();
+
+            if (count($this->schemas) === 0) {
+                throw new InvariantException('Index schema is not specified and there is no active schemas');
+            }
+        } else {
+            $this->schemas = [$schema];
+        }
+    }
+
     private function parseOptionIds(): Set
     {
         $ret = new Set();
+        $ids = $this->option('ids');
+        $ids = is_array($ids) ? $ids : explode(',', get_string($ids) ?? '');
```

Comment:
seems strange that `$ids` allows array without `--ids=*` as option signature 🤔  (and weirder that laravel doesn't enforce `*` when passed as direct argument 🤔 )

Reply:
I don't think supporting that for cli is a good idea... The resulting option would then always be in an array and if we're going to still support comma separated value it also means additional parsing inside the array.

Laravel doesn't seem to do anything with the param values when passed using `Artisan::call`.